### PR TITLE
Support for replacing the TabController, after disposing the old one

### LINKED ISSUE #32428: Replacing the TabController, disposing the old one, fails
Pressing the "Change TabController length" button in this app demonstrates the problem: https://gist.github.com/HansMuller/dd9745558b991f740a9f4f31ddf10923

It happens because the didUpdateWidget methods in TabBar and TabBarView tried to remove their listeners from the old disposed controller.

### PR DESCRIPTION
**Pull request by HansMuller**
Fixes https://github.com/flutter/flutter/issues/32428

### CODE REVIEW

File: packages/flutter/lib/src/material/tab_controller.dart [hunk 1]
```diff
@@ -130,10 +130,9 @@ class TabController extends ChangeNotifier {
   /// animation's value can be [offset] by +/- 1.0 to reflect [TabBarView]
   /// drag scrolling.
   ///
-  /// If length is zero or one, [index] animations don't happen and the value
-  /// of this property is [kAlwaysCompleteAnimation].
-  Animation<double> get animation => _animationController?.view ?? kAlwaysCompleteAnimation;
-  final AnimationController _animationController;
+  /// If the TabController was disposed then return null.
```
Comment (johnsonmh) [inline packages/flutter/lib/src/material/tab_controller.dart]:
```suggestion
  /// If the [TabController] was disposed then return null.
```

File: packages/flutter/test/material/tabs_test.dart [hunk 2]
```diff
@@ -2299,4 +2299,55 @@ void main() {
...
+    expect(find.text('Tab1'), findsOneWidget);
+    expect(find.text('Tab2'), findsNothing);
+
+    await tester.tap(find.text('Change TabController length'));
+    await tester.pumpAndSettle();
+    expect(controller.index, 0);
+    expect(controller.length, 3);
+    expect(find.text('Tab0'), findsOneWidget);
+    expect(find.text('Tab1'), findsOneWidget);
+    expect(find.text('Tab2'), findsOneWidget);
```
Comment (johnsonmh) [inline packages/flutter/test/material/tabs_test.dart]: should you also add an:

```dart
expect(find.text('Tab3'), findsOneWidget);
```

Now that there should be an additional tab?

**Review by johnsonmh — APPROVED** [review]
LGTM

File: packages/flutter/test/material/tabs_test.dart (hunk 2 above)
Comment (HansMuller) [reply packages/flutter/test/material/tabs_test.dart]: I'm changing the length from 2 - 3

**Pull request merged**